## Formative Assessment 6
Statistical Theory

1. Table 1 shows a frequency distribution of grades on a final examination in
college algebra. Find the quartiles of the distribution.

In [62]:
import pandas as pd
import numpy as np
import itertools

In [24]:
table_1 = {
    "Grade":["90-100","80-89","70-79","60-69","50-59","40-49","30-39"],
    "Number of Students": [9,32,43,21,11,3,1]
}
table_1 = pd.DataFrame(table_1)
table_1[["Lower Limit Grade","Upper Limit Grade"]]= table_1["Grade"].str.split("-",expand = True)
table_1.drop(columns=['Grade'],inplace = True)
table_1 = table_1.iloc[:,[1,2,0]]
table_1["Lower Limit Grade"].astype(int)
table_1["Upper Limit Grade"].astype(int)
table_1

Lower Limit Grade Upper Limit Grade  Number of Students
0                90               100                   9
1                80                89                  32
2                70                79                  43
3                60                69                  21
4                50                59                  11
5                40                49                   3
6                30                39                   1

We have 4 quartiles, which are Q_1, Q_2, Q_3, Q_4. Which are defined as the follwing:
1. Q_1 - 25% of the data
2. Q_2 - 50% of the data
3. Q_3 - 75% of the data
4. Q_4 - 90% of the data

Let us first find the cumulative frequency of the given grouped data.

In [26]:
table_1["Cumulative Freq."] = table_1["Number of Students"].cumsum()
table_1

Lower Limit Grade Upper Limit Grade  Number of Students  Cumulative Freq.
0                90               100                   9                 9
1                80                89                  32                41
2                70                79                  43                84
3                60                69                  21               105
4                50                59                  11               116
5                40                49                   3               119
6                30                39                   1               120

Now given that te total is 120 students we can use this to determine the quartiles where each quartile shall be represented by the following equation:
1. Q_1 = 120 * 1/4
2. Q_2 = 120 * 1/2
3. Q_3 = 120 * 2/3
4. Q_4 = 120 * 9/10

In [28]:
quartiles = []
proportions = [1/4,1/2,3/4,9/10]
total_n_students = 120
for prop in proportions:
    quartiles.append(total_n_students*prop)
for i,q in enumerate(quartiles):
    print(f"Q_{i+1} = {q}")

Q_1 = 30.0
Q_2 = 60.0
Q_3 = 90.0
Q_4 = 108.0


Given these intervals it shows the following class intervals for each quarters:
1. Q_1 => 30-39
2. Q_2 => 60-69
3. Q_3 => 90-100	
4. Q_4 => No Class Interval Higher than 108

2. On a final examination in statistics, the mean grade of a group of 150
students was 78 and the standard deviation was 8.0. In algebra, however, the mean final
grade of the group was 73 and the standard deviation was 7.6. In which subject was
there the greater (a) absolute dispersion and (b) relative dispersion?

For absolute dispersion we can look at both the standard deviations of the two subject examinations, for relative dispersion, since there are different N for both subject which would entail that there are different means.

In [35]:
statistics = {
    "mean":78,
    "stdDev":8.0,
}
algebra = {
    "mean":73,
    "stdDev":7.6,
}
statistics["coefVar"] = statistics["stdDev"]/statistics["mean"]
algebra["coefVar"] = algebra["stdDev"]/algebra["mean"]

absoluteDispersion = "statistics" if statistics["stdDev"]>algebra["stdDev"] else "algebra"
relativeDispersion = "statistics" if statistics["coefVar"]>algebra["coefVar"] else "algebra"

absoluteDispersionValue = statistics["stdDev"] if statistics["stdDev"]>algebra["stdDev"] else algebra["stdDev"]
relativeDispersionValue = statistics["coefVar"] if statistics["coefVar"]>algebra["coefVar"] else algebra["coefVar"]

print(f"Greater Absolute Dispersion: {absoluteDispersion}\nwith a value of: {absoluteDispersionValue}")
print(f"\nGreater Relative Dispersion: {relativeDispersion}\nth a value of: {relativeDispersionValue}")

Greater Absolute Dispersion: statistics
with a value of: 8.0

Greater Relative Dispersion: algebra
th a value of: 0.10410958904109588


3.  Prove that the mean and standard deviation of a set of standard scores are
equal to 0 and 1, respectively. Use the following problem to illustrate this: Convert the
set 6, 2, 8, 7, 5 into standard scores.

In [51]:
setOfNums = [6, 2, 8, 7, 5 ]
# to convert to standard scores we must find the z-score where it is given by the formula x-mean/stdDev
meanOfSetOfNums = np.mean(setOfNums)
stdDevOfSetOfNums = np.std(setOfNums)
setOfStandardScores = (np.array(setOfNums) - meanOfSetOfNums)/stdDevOfSetOfNums
print(f"Set of Standard Scores: {setOfStandardScores}")
print(f"Mean of the Set of Standard Scores: {round(np.mean(setOfStandardScores))}")
print(f"Standard Deviation of the Set of Standard Scores: {round(np.std(setOfStandardScores))}")

Set of Standard Scores: [ 0.19425717 -1.74831455  1.16554303  0.6799001  -0.29138576]
Mean of the Set of Standard Scores: 0
Standard Deviation of the Set of Standard Scores: 1


4. Three masses are measured as 20.48, 35.97, and 62.34 g, with standard
deviations of 0.21, 0.46, and 0.54 g, respectively. Find the (a) mean and (b) standard
deviation of the sum of the masses.

standard deviations are the value of the deviations from the mean for each given x.

In [44]:
masses = [20.48, 35.97, 62.34]
standardDevPerX = [0.21,0.46,0.54]
mean_masses = np.mean(masses)
standardDev = ((sum(np.array(standardDevPerX)**2))/len(standardDevPerX))**(1/2)
print(f"Mean of the Masses: {mean_masses}\nStandard deviation of the sum of the masses: {standardDev}")

Mean of the Masses: 39.59666666666667
Standard deviation of the sum of the masses: 0.42712215270731785


5. The credit hour distribution at Metropolitan Technological College is as
follows:

In [55]:
creditHourDistribution = {
    "x":[6,9,12,15,18],
    "p(x)": [0.1,0.2,0.4,0.2,0.1]
}
creditHourDistribution = pd.DataFrame(creditHourDistribution)
creditHourDistribution

x  p(x)
0   6   0.1
1   9   0.2
2  12   0.4
3  15   0.2
4  18   0.1

Find 𝜇 and 𝜎^2. Give the 25 (with replacement) possible samples of size 2, their means,
and their probabilities.

In [58]:
mean = sum(creditHourDistribution["x"]*creditHourDistribution["p(x)"])
variance = (sum(((creditHourDistribution["x"]-mu)**2)*creditHourDistribution["p(x)"]))**2

In [67]:
values = creditHourDistribution["x"].tolist()
probabilities = creditHourDistribution["p(x)"].tolist()

valuesAndProbabs = list(zip(values, probabilities))

In [72]:
samples = list(itertools.product(valuesAndProbabs, repeat=2))
x_1 = []
p__x_1 = []
x_2 = []
p__x_2 = []
mean = []
p__x_1_x_2 = []
sampleTitle = []
for i, s in enumerate(samples):
    val_x1 = s[0][0]
    val_x2 = s[1][0]
    prob_x1 = s[0][1]
    prob_x2 = s[1][1]
    
    meanVal = (val_x1 * prob_x1) + (val_x2 * prob_x2)
    probab_X1_X2 = prob_x1 * prob_x2
    sampleTitleVal = f"Sample {i}"
    
    x_1.append(val_x1)
    p__x_1.append(prob_x1)
    x_2.append(val_x2)
    p__x_2.append(prob_x2)
    mean.append(meanVal)
    p__x_1_x_2.append(probab_X1_X2)
    sampleTitle.append(sampleTitleVal)

results_df = pd.DataFrame({
    'Sample Title': sampleTitle,
    'X1': x_1,
    'P(X1)': p__x_1,
    'X2': x_2,
    'P(X2)': p__x_2,
    'Mean': mean,
    'P(X1, X2)': p__x_1_x_2
})

results_df

Sample Title  X1  P(X1)  X2  P(X2)  Mean  P(X1, X2)
0      Sample 0   6    0.1   6    0.1   1.2       0.01
1      Sample 1   6    0.1   9    0.2   2.4       0.02
2      Sample 2   6    0.1  12    0.4   5.4       0.04
3      Sample 3   6    0.1  15    0.2   3.6       0.02
4      Sample 4   6    0.1  18    0.1   2.4       0.01
5      Sample 5   9    0.2   6    0.1   2.4       0.02
6      Sample 6   9    0.2   9    0.2   3.6       0.04
7      Sample 7   9    0.2  12    0.4   6.6       0.08
8      Sample 8   9    0.2  15    0.2   4.8       0.04
9      Sample 9   9    0.2  18    0.1   3.6       0.02
10    Sample 10  12    0.4   6    0.1   5.4       0.04
11    Sample 11  12    0.4   9    0.2   6.6       0.08
12    Sample 12  12    0.4  12    0.4   9.6       0.16
13    Sample 13  12    0.4  15    0.2   7.8       0.08
14    Sample 14  12    0.4  18    0.1   6.6       0.04
15    Sample 15  15    0.2   6    0.1   3.6       0.02
16    Sample 16  15    0.2   9    0.2   4.8       0.04
17    Sample 17  15    0.2  12    0.4   7.8       0.08
18    Sample 18  15    0.2  15    0.2   6.0       0.04
19    Sample 19  15    0.2  18    0.1   4.8       0.02
20    Sample 20  18    0.1   6    0.1   2.4       0.01
21    Sample 21  18    0.1   9    0.2   3.6       0.02
22    Sample 22  18    0.1  12    0.4   6.6       0.04
23    Sample 23  18    0.1  15    0.2   4.8       0.02
24    Sample 24  18    0.1  18    0.1   3.6       0.01